R/gh_token.R

```diff
@@ -83,7 +83,21 @@
 gh_token <- function(api_url = NULL) {
   api_url <- api_url %||% default_api_url()
   token_env_var <- paste0("GITHUB_PAT_", slugify_url(api_url))
-  get_first_token_found(c(token_env_var, "GITHUB_PAT", "GITHUB_TOKEN"))
+  candidates <- c(
+    token_env_var,
+    if (is_github_dot_com(api_url)) c("GITHUB_PAT", "GITHUB_TOKEN")
```

Comment:
This makes it so we no longer send a GITHUB_PAT to a server other than GitHub.com. I saw this in credentials/gert and I think it's a good idea here too.